### PR TITLE
Update naive-bayes.md

### PR DESCRIPTION
**Pull request by particle1331**
*Description of changes:*

There's a mistake in the formula for calculating naive bayes estimates.
P[x_i, y] is not defined. Only P[i, y] is defined which is the conditional probability p(x_i = 1 | y). 

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1656/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

**Comment by goldmermaid** [comment]
Hey @particle1331 , great catch! I like the way you describe the formula, while it seems to be not a legitimate one... I reconstruct the math and text in the following way:

image

If it makes sense to you, please update this PR as is.


(Attached the latex below.)
```
If we can estimate $\prod_i p(x_i=1  \mid  y)$ for every $i$ and $y$, and save its value in $P_{xy}[i, y]$, here $P_{xy}$ is a $d\times n$ matrix with $n$ being the number of classes and $y\in\{1, \ldots, n\}$, i.e.,


$$ 
P_{xy}[i, y] = 
\begin{cases}
    P_{xy}[i, y] & \text{for } t_i=1\\
    1 - P_{xy}[i, y] & \text{for } t_i = 0
\end{cases}
$$

In addition, we estimate $p(y)$ for every $y$ and save it in $P_y[y]$, with $P_y$ a $n$-length vector. Then for any new example $\mathbf x$, we could compute

$$ 
\begin{aligned}
\hat{y} 
&= \mathrm{argmax}_y \ \prod_{i=1}^d P_y[y] \times P_{xy}[i, y]\\
&= \mathrm{argmax}_y \ \prod_{i=1}^d P_y[y]\ P_{xy}[i, y]^{t_i}\, \left(1 - P_{xy}[i, y]\right)^{1-t_i},
\end{aligned}
$$
:eqlabel:`eq_naive_bayes_estimation`


for any $y$. So our assumption of conditional independence has taken the complexity of our model from an exponential dependence on the number of features $\mathcal{O}(2^dn)$ to a linear dependence, which is $\mathcal{O}(dn)$.
```

**Comment by mli** [comment]
Job d2l-en/PR-1656/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

**Comment by particle1331** [comment]
@goldmermaid This looks awesome! Will do, thanks.

**Comment by mli** [comment]
Job d2l-en/PR-1656/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

**Comment by goldmermaid** [comment]
LGMT! Thanks for your contribution @particle1331 !

**Review by goldmermaid — APPROVED** [review]

**Comment by particle1331** [comment]
@goldmermaid Hi. I'm sorry I wasn't able to push the changes outlined in the comment last night. Please check the recent changes if you have time. Thanks!

**Comment by mli** [comment]
Job d2l-en/PR-1656/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

**Comment by mli** [comment]
Job d2l-en/PR-1656/5 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

**Comment by mli** [comment]
Job d2l-en/PR-1656/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

**Comment by mli** [comment]
Job d2l-en/PR-1656/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

File: chapter_appendix-mathematics-for-deep-learning/naive-bayes.md [hunk 1]
```diff
@@ -168,11 +168,26 @@ By itself, this expression does not get us any further. We still must estimate r
...
+\end{cases}
+$$
+
+In addition, we estimate $p(y)$ for every $y$ and save it in $P_y[y]$, with $P_y$ a $n$-length vector. Then, for any new example $\mathbf t = (t_1, t_2, \ldots, t_d)$, we could compute
+
+$$\begin{aligned}
+\hat{y} 
+&= \mathrm{argmax}_y \ p(y)\prod_{i=1}^d   p(x_t = t_i \mid y)\\
+&= \mathrm{argmax}_y \ P_y[y]\prod_{i=1}^d \ P_{xy}[i, y]^{t_i}\, \left(1 - P_{xy}[i, y]\right)^{1-t_i}
+\end{aligned}$$
```
Comment (goldmermaid) [inline chapter_appendix-mathematics-for-deep-learning/naive-bayes.md:187]: Please seperate the "$$" to a new line. Or the ":eqlabel:`eq_naive_bayes_estimation`" won't render properly.

Comment (goldmermaid) [reply chapter_appendix-mathematics-for-deep-learning/naive-bayes.md:187]: i.e., 
```
$$
\begin{aligned}

...

\end{aligned}
$$
:eqlabel:`eq_naive_bayes_estimation`
```

**Comment by mli** [comment]
Job d2l-en/PR-1656/9 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

**Comment by mli** [comment]
Job d2l-en/PR-1656/10 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

**Comment by astonzhang** [comment]
@goldmermaid  does this PR look good to you now?

File: chapter_appendix-mathematics-for-deep-learning/naive-bayes.md [hunk 2]
```diff
@@ -168,11 +168,29 @@ By itself, this expression does not get us any further. We still must estimate r
...
+
+In addition, we estimate $p(y)$ for every $y$ and save it in $P_y[y]$, with $P_y$ a $n$-length vector. Then for any new example $\mathbf x$, we could compute
+
+$$ 
+\begin{aligned}
+\hat{y} 
+&= \mathrm{argmax}_y \ \prod_{i=1}^d P_y[y] \times P_{xy}[i, y]\\
+&= \mathrm{argmax}_y \ \prod_{i=1}^d P_y[y]\ P_{xy}[i, y]^{t_i}\, \left(1 - P_{xy}[i, y]\right)^{1-t_i},
+\end{aligned}
+$$
```
Comment (goldmermaid) [inline chapter_appendix-mathematics-for-deep-learning/naive-bayes.md]: Hey @particle1331 , apology for a bit delay. Here is the reason why this "eqlabel" is not render well: 

"$$\begin{aligned} ... \end{aligned}$$" needs to be on one single line. 

Please check (search `\begin{aligned}`) in this example: https://github.com/d2l-ai/d2l-en/blob/master/chapter_linear-networks/linear-regression.md. 

Let me know if you are still having trouble solving it.

**Comment by mli** [comment]
Job d2l-en/PR-1656/11 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

**Comment by mli** [comment]
Job d2l-en/PR-1656/12 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

**Comment by mli** [comment]
Job d2l-en/PR-1656/13 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1656/

File: chapter_appendix-mathematics-for-deep-learning/naive-bayes.md (hunk 2 above)
Comment (particle1331) [reply chapter_appendix-mathematics-for-deep-learning/naive-bayes.md]: Hi! Sorry for the delay as well. Thank you for pointing me towards this solution. 
It worked. :=) 

http://preview.d2l.ai/d2l-en/PR-1656/chapter_appendix-mathematics-for-deep-learning/naive-bayes.html

**Comment by goldmermaid** [comment]
LGTM! Fantastic work @particle1331 !

**Pull request merged**